sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/FaultInjectionWithAvailabilityStrategyTests.java

```diff
@@ -1531,48 +1528,262 @@ public void writeAfterCreation(
+            queryReturnsTotalRecordCountCore(query, params, 1, true);
+
+        BiConsumer<CosmosResponseWrapper, Long>  validateExpectedRecordCount = (response, expectedRecordCount) -> {
+            if (expectedRecordCount != null) {
+                assertThat(response).isNotNull();
+                assertThat(response.getTotalRecordCount()).isNotNull();
+                assertThat(response.getTotalRecordCount()).isEqualTo(expectedRecordCount);
+            }
         };
 
+        Consumer<CosmosResponseWrapper> validateEmptyResults =
+            (response) -> validateExpectedRecordCount.accept(response, 0L);
+
+        Consumer<CosmosResponseWrapper> validateExactlyOneRecordReturned =
+            (response) -> validateExpectedRecordCount.accept(response, 1L);
+
+        Consumer<CosmosResponseWrapper> validateAllRecordsSameIdReturned =
+            (response) -> validateExpectedRecordCount.accept(
+                response,
+                1L + ENOUGH_DOCS_OTHER_PK_TO_HIT_EVERY_PARTITION);
+
+        Consumer<CosmosResponseWrapper> validateAllRecordsSamePartitionReturned =
+            (response) -> validateExpectedRecordCount.accept(
+                response,
+                1L + ENOUGH_DOCS_SAME_PK_TO_EXCEED_PAGE_SIZE);
+
         Function<ItemOperationInvocationParameters, String> singlePartitionQueryGenerator = (params) ->
-            "SELECT * FROM c WHERE c.id = '"
-                + params.idAndPkValuePair.getLeft()
-                + "' AND c.mypk = '"
+            "SELECT * FROM c WHERE c.mypk = '"
                 +  params.idAndPkValuePair.getRight()
                 + "'";
 
+        Function<ItemOperationInvocationParameters, String> singlePartitionWithAggregatesAndOrderByQueryGenerator = (params) ->
+            "SELECT DISTINCT c.id FROM c WHERE c.mypk = '"
+                +  params.idAndPkValuePair.getRight()
+                + "' ORDER BY c.id";
+
+        Function<ItemOperationInvocationParameters, String> singlePartitionEmptyResultQueryGenerator = (params) ->
+            "SELECT * FROM c WHERE c.mypk = '"
+                +  params.idAndPkValuePair.getRight()
+                + "' and c.id = 'NotExistingId'";
+
+        Function<ItemOperationInvocationParameters, String> crossPartitionQueryGenerator = (params) ->
+            "SELECT * FROM c WHERE CONTAINS (c.id, '"
+                + params.idAndPkValuePair.getLeft()
+                + "')";
+
+        Function<ItemOperationInvocationParameters, String> crossPartitionWithAggregatesAndOrderByQueryGenerator = (params) ->
+            "SELECT DISTINCT c.id FROM c WHERE CONTAINS (c.id, '"
+                + params.idAndPkValuePair.getLeft()
+                + "')";
+
+        Function<ItemOperationInvocationParameters, String> crossPartitionEmptyResultQueryGenerator = (params) ->
+            "SELECT * FROM c WHERE CONTAINS (c.id, 'NotExistingId')";
+
```

Comment:
Agreed for readMany and readAll - will add some coverage for those

All kind of query flavors seems too much here - I have intentionally covered distinct with order by because that uses a different code path in query pipeline - believe that is a reasonable compromise

Reply:
Will add the tests for readAll and readMany in separate PR.